### PR TITLE
Add match

### PR DESCRIPTION
**Pull request by yuriyoleynikov**
Addresses #37

### CODE REVIEW

File: src/YoTennis/Controllers/MatchController.cs [hunk 1]
```diff
@@ -28,6 +28,47 @@ public MatchController(UserManager<ApplicationUser> userManager, IMatchListServi
             _matchListService = matchListService;
         }
 
+        public async Task<IActionResult> Add()
```
Comment (Igorbek) [inline src/YoTennis/Controllers/MatchController.cs]: I'd suggest to rename to something like `AddPastMatch`

File: src/YoTennis/Controllers/MatchController.cs [hunk 2]
```diff
@@ -28,6 +28,47 @@ public MatchController(UserManager<ApplicationUser> userManager, IMatchListServi
...
+                        Score = new Score {
+                            FirstPlayer = addMatchViewModel.FirstPlayerSet1, SecondPlayer = addMatchViewModel.SecondPlayerSet1 }
+                    },
+                    new SetModel {
+                        Score = new Score {
+                            FirstPlayer = addMatchViewModel.FirstPlayerSet2, SecondPlayer = addMatchViewModel.SecondPlayerSet2 }
+                    },
+                    new SetModel {
+                        Score = new Score {
+                            FirstPlayer = addMatchViewModel.FirstPlayerSet3, SecondPlayer = addMatchViewModel.SecondPlayerSet3 }
```
Comment (Igorbek) [inline src/YoTennis/Controllers/MatchController.cs]: Is it always going to be 3 sets? How would you detect that is more?

File: src/YoTennis/Models/AddMatchViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace YoTennis.Models
+{
+    public class AddMatchViewModel
```
Comment (Igorbek) [inline src/YoTennis/Models/AddMatchViewModel.cs]: also like `AddPostMatchViewModel`

File: src/YoTennis/Models/GameHandler.cs [hunk 4]
```diff
@@ -155,7 +155,23 @@ public void AddEvent(GameEvent gameEvent)
...
+                        
+            CurrentState.FirstPlayer = gameEvent.FirstPlayer;
+            CurrentState.SecondPlayer = gameEvent.SecondPlayer;
+            CurrentState.MatchStartedAt = gameEvent.Date;
+
+            CurrentState.FirstPlayerUserId = gameEvent.FirstPlayerUserId;
+            CurrentState.SecondPlayerUserId = gameEvent.SecondPlayerUserId;
+
+            CurrentState.Sets = gameEvent.MatchScore;
+
```
Comment (Igorbek) [inline src/YoTennis/Models/GameHandler.cs]: the handler does not change state, so it will still be `NotStarted`

**Review by Igorbek — CHANGES_REQUESTED** [review]
In addition to comments, please add tests that covers past match importing

File: src/YoTennis/Controllers/MatchController.cs (hunk 1 above)
Comment (yuriyoleynikov) [reply src/YoTennis/Controllers/MatchController.cs]: Done

File: src/YoTennis/Models/AddMatchViewModel.cs (hunk 3 above)
Comment (yuriyoleynikov) [reply src/YoTennis/Models/AddMatchViewModel.cs]: Done

File: src/YoTennis/Models/GameHandler.cs (hunk 4 above)
Comment (yuriyoleynikov) [reply src/YoTennis/Models/GameHandler.cs]: Done

Comment (yuriyoleynikov) [reply src/YoTennis/Models/GameHandler.cs]: This was hard!

**Review by Igorbek — APPROVED** [review]

**Pull request merged**